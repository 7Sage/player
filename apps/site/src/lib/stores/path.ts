import { derived } from 'svelte/store';

import { page } from '$app/stores';

export const isReactPath = derived(page, ($page) => /\/react\/?/.test($page.url.pathname));

export const isComponentsPath = derived(page, ($page) =>
  /\/components\/?/.test($page.url.pathname),
);

export const isApiPath = derived(page, ($page) => /\/api\/?/.test($page.url.pathname));
