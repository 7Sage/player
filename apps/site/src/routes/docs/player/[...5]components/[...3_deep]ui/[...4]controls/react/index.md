---
title: Controls Docs (React)
---

<script>
import Docs from '../_Docs.md';
</script>

<Docs>

```jsx copyHighlight|slot=styling{4-8}
<Media>
  <Video>{/* ... */}</Video>

  <div className="media-controls-container">
    <div className="media-controls"></div>
    <div className="media-controls"></div>
    <div className="media-controls"></div>
  </div>
</Media>
```

</Docs>
