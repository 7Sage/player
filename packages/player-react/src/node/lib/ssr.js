import '@vidstack/player/shims/install-ssr.js';

import { render } from '@lit-labs/ssr';
import { html } from 'lit';
import { unsafeHTML } from 'lit/directives/unsafe-html.js';

async function main() {
  const { html: reactHTML, tagNames } = /** @type {{ html: string; tagNames: string; }} */ (
    process.env
  );

  const tags = JSON.parse(tagNames);

  for (const tag of tags) {
    await import(`@vidstack/player/define/${tag}.js`);
  }

  const template = html`${unsafeHTML(reactHTML)}`;
  const ssrHTML = trimOuterMarkers(iterableToString(render(template)));

  process.stdout.write(JSON.stringify({ html: ssrHTML }));
}

function trimOuterMarkers(renderedContent) {
  return renderedContent
    .replace(/^((<!--[^<>]*-->)|(<\?>)|\s)+/, '')
    .replace(/((<!--[^<>]*-->)|(<\?>)|\s)+$/, '');
}

const iterableToString = (iterable) => {
  let s = '';
  for (const i of iterable) s += i;
  return s;
};

main();
