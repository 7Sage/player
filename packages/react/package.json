{
  "name": "@vidstack/react",
  "version": "1.12.0",
  "description": "UI component library for building high-quality, accessible video and audio experiences on the web.",
  "license": "MIT",
  "type": "module",
  "types": "index.d.ts",
  "sideEffects": false,
  "engines": {
    "node": ">=18"
  },
  "scripts": {
    "analyze": "maverick analyze \"src/**/*.tsx\" --framework react --project tsconfig.build.json",
    "dev": "pnpm clean && pnpm build:icons && pnpm build:types && pnpm watch",
    "build": "pnpm build:icons && tsx .scripts/build.ts",
    "build:icons": "node .scripts/build-icons.js",
    "build:types": "tsc -p tsconfig.build.json",
    "sandbox": "node ../../.scripts/sandbox.js",
    "sandbox:build": "vite build sandbox",
    "types": "pnpm build:icons && pnpm build:types && rollup -c",
    "clean": "rimraf dist-npm",
    "format": "prettier src --write --loglevel warn",
    "watch": "pnpm watch:types & pnpm run build -w",
    "watch:types": "pnpm run build:types -w & rollup -c -w"
  },
  "peerDependencies": {
    "@types/react": "^18.0.0",
    "react": "^18.0.0"
  },
  "dependencies": {
    "@floating-ui/dom": "^1.4.4",
    "media-captions": "^1.0.1"
  },
  "devDependencies": {
    "@maverick-js/cli": "0.43.1",
    "@types/fs-extra": "^11.0.1",
    "@types/react": "^18.0.0",
    "@types/react-dom": "^18.0.0",
    "@vitejs/plugin-react": "^4.2.0",
    "esbuild": "^0.23.0",
    "fs-extra": "^11.0.0",
    "maverick.js": "0.43.1",
    "media-icons": "^1.1.4",
    "react": "^18.0.0",
    "react-dom": "^18.0.0",
    "remotion": "^4.0.76",
    "rimraf": "^3.0.0",
    "rollup": "^4.0.0",
    "rollup-plugin-dts": "^6.0.0",
    "tslib": "^2.6.0",
    "tsx": "^4.16.2",
    "type-fest": "^3.8.0",
    "typescript": "^5.3.0",
    "vidstack": "workspace:*",
    "vite": "^5.0.0"
  },
  "contributors": [
    "Rahim Alwer <rahim.alwer@gmail.com>"
  ],
  "repository": {
    "type": "git",
    "url": "https://github.com/vidstack/player.git",
    "directory": "packages/react"
  },
  "bugs": {
    "url": "https://github.com/vidstack/player/issues"
  },
  "exports": {
    ".": {
      "types": "./index.d.ts",
      "worker": "./server/vidstack.js",
      "development": "./dev/vidstack.js",
      "default": "./prod/vidstack.js"
    },
    "./icons": {
      "types": "./icons.d.ts",
      "default": "./prod/vidstack-icons.js"
    },
    "./player/remotion": {
      "types": "./player/remotion.d.ts",
      "worker": "./server/vidstack-remotion.js",
      "development": "./dev/player/vidstack-remotion.js",
      "default": "./prod/player/vidstack-remotion.js"
    },
    "./player/layouts/default": {
      "types": "./player/layouts/default.d.ts",
      "worker": "./server/vidstack-default-layout.js",
      "development": "./dev/player/vidstack-default-layout.js",
      "default": "./prod/player/vidstack-default-layout.js"
    },
    "./player/layouts/plyr": {
      "types": "./player/layouts/plyr.d.ts",
      "worker": "./server/vidstack-plyr-layout.js",
      "development": "./dev/player/vidstack-plyr-layout.js",
      "default": "./prod/player/vidstack-plyr-layout.js"
    },
    "./player/styles/*": "./player/styles/*",
    "./types/*": "./types/*",
    "./dom.d.ts": "./dom.d.ts",
    "./google-cast.d.ts": "./google-cast.d.ts",
    "./package.json": "./package.json",
    "./tailwind.cjs": {
      "types": "./tailwind.d.cts",
      "default": "./tailwind.cjs"
    },
    "./analyze.json": {
      "types": "./analyze.json.d.ts",
      "default": "./analyze.json"
    }
  },
  "publishConfig": {
    "access": "public"
  },
  "keywords": [
    "accessible",
    "analytics",
    "audio",
    "customizable",
    "dailymotion",
    "elements",
    "extensible",
    "file",
    "framework-agnostic",
    "headless",
    "hls",
    "html5",
    "jwplayer",
    "media",
    "plyr",
    "player",
    "react",
    "spec-compliant",
    "universal",
    "uploader",
    "vds",
    "video",
    "videojs",
    "vidstack",
    "vime",
    "vimeo",
    "web-component",
    "tailwind",
    "universal",
    "youtube"
  ]
}
