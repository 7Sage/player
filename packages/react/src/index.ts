export * from './components';
export * from './use-media-player';
export * from './use-media-provider';
export * from './use-media-remote';
export * from './use-media-store';
