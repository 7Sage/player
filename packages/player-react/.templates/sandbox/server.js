import React from '@vitejs/plugin-react';
import express from 'express';
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __dirname = path.dirname(fileURLToPath(import.meta.url));

export async function createServer(root = __dirname) {
  const resolve = (p) => path.resolve(__dirname, p);

  const app = express();

  /**
   * @type {import('vite').ViteDevServer}
   */
  const vite = await (
    await import('vite')
  ).createServer({
    root,
    logLevel: isTest ? 'error' : 'info',
    server: {
      middlewareMode: 'ssr',
      watch: {
        usePolling: true,
        interval: 100,
      },
    },
    plugins: [React()],
  });

  // use vite's connect instance as middleware
  app.use(vite.middlewares);

  app.use('*', async (req, res) => {
    try {
      const url = req.originalUrl;

      let template = fs.readFileSync(resolve('index.html'), 'utf-8');
      template = await vite.transformIndexHtml(url, template);

      const render = (await vite.ssrLoadModule('/entry-server.tsx')).render;
      const appHtml = render(url);
      const html = template.replace(`<!--app-html-->`, appHtml);

      res.status(200).set({ 'Content-Type': 'text/html' }).end(html);
    } catch (e) {
      vite.ssrFixStacktrace(e);
      console.log(e.stack);
      res.status(500).end(e.stack);
    }
  });

  return { app, vite };
}

const isTest = process.env.NODE_ENV === 'test' || !!process.env.VITE_TEST_BUILD;

if (!isTest) {
  createServer().then(({ app }) =>
    app.listen(3100, () => {
      console.log('http://localhost:3100');
    }),
  );
}
