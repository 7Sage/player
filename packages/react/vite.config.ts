import path from 'node:path';

import react from '@vitejs/plugin-react';
import { defineConfig } from 'vite';

const localPath = path.resolve('../vidstack/src/index.ts');

// https://vitejs.dev/config/
export default defineConfig({
  define: {
    __DEV__: 'true',
    __SERVER__: 'false',
  },
  plugins: [react()],
  resolve: {
    alias: {
      'vidstack/local': localPath,
    },
  },
  optimizeDeps: {
    exclude: [
      'vidstack',
      'vidstack/local',
      localPath,
      'maverick.js',
      'maverick.js/react',
      'media-icons',
      'media-captions',
    ],
  },
});
