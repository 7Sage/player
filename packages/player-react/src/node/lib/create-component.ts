/**
 * Adapted from https://github.com/lit/lit/blob/main/packages/labs/react/src/create-component.ts.
 */

import { camelToKebabCase } from '@vidstack/foundation';
import { execSync } from 'child_process';
import path from 'path';
import * as ReactModule from 'react';
import { renderToString } from 'react-dom/server';
import { fileURLToPath } from 'url';

let ssrInProgress = false;

const tags = new Set<string>();
const blacklist = new Set(['children', 'localName', 'ref', 'style', 'className']);

const ssrFile = path.resolve(path.dirname(fileURLToPath(import.meta.url)), './exec-ssr.js');

const camelCaseRE = /[a-z][A-Z]/g;

export const createComponent = (
  React: typeof ReactModule,
  tagName: string,
  { skipSSR = false } = {},
) => {
  const Component = React.Component;
  const createElement = React.createElement;

  class ReactComponent extends Component {
    override render() {
      tags.add(tagName);

      const props: any = { ...this.props };
      delete props.__forwardedRef;

      for (const prop of Object.keys(props)) {
        if (prop.startsWith('_') || blacklist.has(prop) || typeof props[prop] === 'function') {
          continue;
        }

        if (typeof props[prop] === 'object') {
          props[prop] = JSON.stringify(props[prop]);
        }

        if (camelCaseRE.test(prop)) {
          props[camelToKebabCase(prop)] = props[prop];
          delete props[prop];
        }
      }

      if (!ssrInProgress && !skipSSR) {
        ssrInProgress = true;

        const openTag = `<${tagName}>`;
        const closeTag = `</${tagName}>`;
        const html = `${openTag}${renderToString(props.children)}${closeTag}`;

        try {
          const ssrResult = execSync(
            `node --unhandled-rejections=strict --abort-on-uncaught-exception ${ssrFile}`,
            {
              env: { ...process.env, html, tagNames: JSON.stringify(Array.from(tags)) },
            },
          ).toString();

          props.dangerouslySetInnerHTML = {
            __html: ssrResult.slice(openTag.length, -closeTag.length),
          };

          delete props.children;
        } catch (e) {
          console.error(e);
        }

        ssrInProgress = false;
      }

      return createElement(tagName, props);
    }
  }

  const ForwardedComponent = React.forwardRef<any, any>((props, ref) =>
    createElement(ReactComponent, { ...props, __forwardedRef: ref }, props?.children),
  );

  return ForwardedComponent;
};
