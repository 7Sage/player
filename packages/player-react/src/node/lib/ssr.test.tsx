import '@vidstack/foundation/shims/install-ssr.js';

import fs from 'fs';
import path from 'path';
import * as React from 'react';
import { renderToString } from 'react-dom/server';

import { createComponent } from './create-component';
import { ssr } from './ssr.js';

const tags: string[] = [];
const elementsPath = path.resolve(process.cwd(), 'node_modules/@vidstack/player/elements.json');
const elements = JSON.parse(fs.readFileSync(elementsPath, { encoding: 'utf-8' }));
tags.push(...elements.components.map((c) => c.tagName));

for (const tag of tags) {
  it(`should SSR ${tag}`, async () => {
    const Component = createComponent(React, tag, { skipSSR: true });
    const reactHTML = renderToString(<Component />);
    const html = await ssr(reactHTML, [tag]);
    expect(html).to.matchSnapshot();
  });
}
