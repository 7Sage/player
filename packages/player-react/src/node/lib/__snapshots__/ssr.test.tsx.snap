// Vitest Snapshot v1

exports[`should SSR vds-aspect-ratio 1`] = `"<vds-aspect-ratio><template shadowroot=\\"open\\"><style>:host{display:block}:host([hidden]){display:none}.container{position:relative;width:100%;height:0;padding-bottom:min(max(var(--vds-aspect-ratio-min-height),var(--vds-aspect-ratio-percent)),var(--vds-aspect-ratio-max-height))}slot{display:block;position:absolute;top:0;left:0;width:100%;height:100%}::slotted(*){--vds-video-width:100%;--vds-video-height:100%;width:100%;height:100%}</style><!--lit-part xwe/eRkxSXY=--><div class=\\"container\\" style=\\"--vds-aspect-ratio-percent:50%;--vds-aspect-ratio-min-height:150px;--vds-aspect-ratio-max-height:100vh;\\"><!--lit-node 0--><slot></slot></div><!--/lit-part--></template></vds-aspect-ratio>"`;

exports[`should SSR vds-audio 1`] = `"<vds-audio><template shadowroot=\\"open\\"><style>:host{display:inline-block}:host([hidden]){display:none}</style><!--lit-part RKsLzNR31Mg=--><slot ><!--lit-node 0--></slot><!--/lit-part--></template></vds-audio>"`;

exports[`should SSR vds-fullscreen-button 1`] = `"<vds-fullscreen-button><template shadowroot=\\"open\\"><style>:host{display:table;contain:content;user-select:none;cursor:pointer;-webkit-user-select:none;-webkit-tap-highlight-color:transparent}:host([hidden]){display:none}:host(:focus){outline:0}:host(:focus-visible){outline:1px auto Highlight;outline:1px auto -webkit-focus-ring-color}:host(.focus-visible){outline:1px auto Highlight;outline:1px auto -webkit-focus-ring-color}</style><!--lit-part Pz0gobCCM4E=--><slot></slot><!--/lit-part--></template></vds-fullscreen-button>"`;

exports[`should SSR vds-gesture 1`] = `"<vds-gesture><template shadowroot=\\"open\\"><style>:host{display:block;contain:content;z-index:0;opacity:0;visibility:hidden;pointer-events:none!important}:host([hidden]){display:none}</style><!--lit-part--><!--/lit-part--></template></vds-gesture>"`;

exports[`should SSR vds-hls 1`] = `"<vds-hls><template shadowroot=\\"open\\"><style>:host{display:inline-block;background-color:var(--vds-video-bg-color,#000)}:host([hidden]){display:none}::slotted(video:not([width])){width:var(--vds-video-width,100%)}::slotted(video:not([height])){height:var(--vds-video-height,auto)}</style><!--lit-part RKsLzNR31Mg=--><slot ><!--lit-node 0--></slot><!--/lit-part--></template></vds-hls>"`;

exports[`should SSR vds-media 1`] = `"<vds-media><template shadowroot=\\"open\\"><style>:host{display:inline-block;position:relative;contain:content}:host([hidden]){display:none}</style><!--lit-part Pz0gobCCM4E=--><slot></slot><!--/lit-part--></template></vds-media>"`;

exports[`should SSR vds-media-sync 1`] = `"<vds-media-sync><template shadowroot=\\"open\\"><style>:host{display:contents}</style><!--lit-part Pz0gobCCM4E=--><slot></slot><!--/lit-part--></template></vds-media-sync>"`;

exports[`should SSR vds-media-visibility 1`] = `"<vds-media-visibility><template shadowroot=\\"open\\"><style>:host{display:contents}</style><!--lit-part Pz0gobCCM4E=--><slot></slot><!--/lit-part--></template></vds-media-visibility>"`;

exports[`should SSR vds-mute-button 1`] = `"<vds-mute-button><template shadowroot=\\"open\\"><style>:host{display:table;contain:content;user-select:none;cursor:pointer;-webkit-user-select:none;-webkit-tap-highlight-color:transparent}:host([hidden]){display:none}:host(:focus){outline:0}:host(:focus-visible){outline:1px auto Highlight;outline:1px auto -webkit-focus-ring-color}:host(.focus-visible){outline:1px auto Highlight;outline:1px auto -webkit-focus-ring-color}</style><!--lit-part Pz0gobCCM4E=--><slot></slot><!--/lit-part--></template></vds-mute-button>"`;

exports[`should SSR vds-play-button 1`] = `"<vds-play-button><template shadowroot=\\"open\\"><style>:host{display:table;contain:content;user-select:none;cursor:pointer;-webkit-user-select:none;-webkit-tap-highlight-color:transparent}:host([hidden]){display:none}:host(:focus){outline:0}:host(:focus-visible){outline:1px auto Highlight;outline:1px auto -webkit-focus-ring-color}:host(.focus-visible){outline:1px auto Highlight;outline:1px auto -webkit-focus-ring-color}</style><!--lit-part Pz0gobCCM4E=--><slot></slot><!--/lit-part--></template></vds-play-button>"`;

exports[`should SSR vds-poster 1`] = `"<vds-poster><template shadowroot=\\"open\\"><style>:host{display:block;contain:content;pointer-events:none;object-fit:cover;box-sizing:border-box}:host([hidden]){display:none}img{display:block;width:100%;height:100%;pointer-events:none;object-fit:inherit;object-position:inherit;user-select:none;-webkit-user-select:none;box-sizing:border-box}</style><!--lit-part AEQfBc2t+jw=--><img part=\\"img\\"    ><!--lit-node 0--><!--/lit-part--></template></vds-poster>"`;

exports[`should SSR vds-slider 1`] = `"<vds-slider aria-valuemin=\\"0\\" aria-valuenow=\\"50\\" aria-valuemax=\\"100\\" aria-valuetext=\\"50%\\"><template shadowroot=\\"open\\"><style>*{box-sizing:border-box;touch-action:none}:host{display:block;contain:layout;user-select:none;-webkit-user-select:none;-webkit-tap-highlight-color:transparent}:host([hidden]){display:none}:host(:focus){outline:0}:host(:focus-visible){outline:1px auto Highlight;outline:1px auto -webkit-focus-ring-color}:host(.focus-visible){outline:1px auto Highlight;outline:1px auto -webkit-focus-ring-color}</style><!--lit-part BRUAAAUVAAA=--><!--lit-part Pz0gobCCM4E=--><slot></slot><!--/lit-part--><?><!--/lit-part--></template></vds-slider>"`;

exports[`should SSR vds-slider-value-text 1`] = `"<vds-slider-value-text><template shadowroot=\\"open\\"><style>:host{display:inline-block;contain:content}:host([hidden]){display:none}</style><!--lit-part BRUAAAUVAAA=--><!--lit-part-->0<!--/lit-part--><?><!--/lit-part--></template></vds-slider-value-text>"`;

exports[`should SSR vds-slider-video 1`] = `"<vds-slider-video><template shadowroot=\\"open\\"><style>:host{display:inline-block;contain:content}:host([hidden]){display:none}video{display:block;width:100%;height:auto}</style><!--lit-part tEqLJ28ErIQ=--><video part=\\"video\\" muted playsinline preload=\\"auto\\"    ><!--lit-node 0--></video><!--/lit-part--></template></vds-slider-video>"`;

exports[`should SSR vds-time 1`] = `"<vds-time><template shadowroot=\\"open\\"><style>:host{display:inline-block;contain:content}:host([hidden]){display:none}</style><!--lit-part BRUAAAUVAAA=--><!--lit-part-->0:00<!--/lit-part--><?><!--/lit-part--></template></vds-time>"`;

exports[`should SSR vds-time-slider 1`] = `"<vds-time-slider aria-valuemin=\\"0%\\" aria-valuenow=\\"0%\\" aria-valuemax=\\"100%\\" aria-valuetext=\\"0 seconds out of 0 seconds\\"><template shadowroot=\\"open\\"><style>*{box-sizing:border-box;touch-action:none}:host{display:block;contain:layout;user-select:none;-webkit-user-select:none;-webkit-tap-highlight-color:transparent}:host([hidden]){display:none}:host(:focus){outline:0}:host(:focus-visible){outline:1px auto Highlight;outline:1px auto -webkit-focus-ring-color}:host(.focus-visible){outline:1px auto Highlight;outline:1px auto -webkit-focus-ring-color}</style><!--lit-part BRUAAAUVAAA=--><!--lit-part Pz0gobCCM4E=--><slot></slot><!--/lit-part--><?><!--/lit-part--></template></vds-time-slider>"`;

exports[`should SSR vds-toggle-button 1`] = `"<vds-toggle-button><template shadowroot=\\"open\\"><style>:host{display:table;contain:content;user-select:none;cursor:pointer;-webkit-user-select:none;-webkit-tap-highlight-color:transparent}:host([hidden]){display:none}:host(:focus){outline:0}:host(:focus-visible){outline:1px auto Highlight;outline:1px auto -webkit-focus-ring-color}:host(.focus-visible){outline:1px auto Highlight;outline:1px auto -webkit-focus-ring-color}</style><!--lit-part Pz0gobCCM4E=--><slot></slot><!--/lit-part--></template></vds-toggle-button>"`;

exports[`should SSR vds-video 1`] = `"<vds-video><template shadowroot=\\"open\\"><style>:host{display:inline-block;background-color:var(--vds-video-bg-color,#000)}:host([hidden]){display:none}::slotted(video:not([width])){width:var(--vds-video-width,100%)}::slotted(video:not([height])){height:var(--vds-video-height,auto)}</style><!--lit-part RKsLzNR31Mg=--><slot ><!--lit-node 0--></slot><!--/lit-part--></template></vds-video>"`;

exports[`should SSR vds-volume-slider 1`] = `"<vds-volume-slider aria-valuemin=\\"0\\" aria-valuenow=\\"100\\" aria-valuemax=\\"100\\" aria-valuetext=\\"100%\\"><template shadowroot=\\"open\\"><style>*{box-sizing:border-box;touch-action:none}:host{display:block;contain:layout;user-select:none;-webkit-user-select:none;-webkit-tap-highlight-color:transparent}:host([hidden]){display:none}:host(:focus){outline:0}:host(:focus-visible){outline:1px auto Highlight;outline:1px auto -webkit-focus-ring-color}:host(.focus-visible){outline:1px auto Highlight;outline:1px auto -webkit-focus-ring-color}</style><!--lit-part BRUAAAUVAAA=--><!--lit-part Pz0gobCCM4E=--><slot></slot><!--/lit-part--><?><!--/lit-part--></template></vds-volume-slider>"`;
